packages/amplify-graphql-model-transformer/src/resources/rds-model-resource-generator.ts

```diff
@@ -23,92 +23,124 @@ import { ModelResourceGenerator } from './model-resource-generator';
-      const dbType = dbTypeOverride ?? getImportedRDSType(context.modelToDatasourceMap);
-      const engine = getEngineFromDBType(dbType);
-      const secretEntry = context.datasourceSecretParameterLocations.get(dbType);
-      const {
-        AmplifySQLLayerNotificationTopicAccount,
-        AmplifySQLLayerNotificationTopicName,
-        SQLLambdaDataSourceLogicalID,
-        SQLLambdaIAMRoleLogicalID,
-        SQLLambdaLogicalID,
-        SQLPatchingLambdaIAMRoleLogicalID,
-        SQLPatchingLambdaLogicalID,
-        SQLPatchingSubscriptionLogicalID,
-        SQLPatchingTopicLogicalID,
-        SQLStackName,
-      } = ResourceConstants.RESOURCES;
-      const lambdaRoleScope = context.stackManager.getScopeFor(SQLLambdaIAMRoleLogicalID, SQLStackName);
-      const lambdaScope = context.stackManager.getScopeFor(SQLLambdaLogicalID, SQLStackName);
-
-      const layerVersionArn = resolveLayerVersion(lambdaScope, context);
-
-      const role = createRdsLambdaRole(
-        context.resourceHelper.generateIAMRoleName(SQLLambdaIAMRoleLogicalID),
-        lambdaRoleScope,
-        secretEntry as RDSConnectionSecrets,
-      );
-
-      const lambda = createRdsLambda(
-        lambdaScope,
-        context.api,
-        role,
-        layerVersionArn,
-        {
-          engine: engine,
-          username: secretEntry?.username ?? '',
-          password: secretEntry?.password ?? '',
-          host: secretEntry?.host ?? '',
-          port: secretEntry?.port ?? '',
-          database: secretEntry?.database ?? '',
-        },
-        context.sqlLambdaVpcConfig,
-        context.sqlLambdaProvisionedConcurrencyConfig,
-      );
-
-      const patchingLambdaRoleScope = context.stackManager.getScopeFor(SQLPatchingLambdaIAMRoleLogicalID, SQLStackName);
-      const patchingLambdaScope = context.stackManager.getScopeFor(SQLPatchingLambdaLogicalID, SQLStackName);
-      const patchingLambdaRole = createRdsPatchingLambdaRole(
-        context.resourceHelper.generateIAMRoleName(SQLPatchingLambdaIAMRoleLogicalID),
-        patchingLambdaRoleScope,
-        lambda.functionArn,
-      );
-
-      // eslint-disable-next-line @typescript-eslint/no-unused-vars
-      const patchingLambda = createRdsPatchingLambda(patchingLambdaScope, context.api, patchingLambdaRole, {
-        LAMBDA_FUNCTION_ARN: lambda.functionArn,
-      });
+  /**
+   * Generates the AWS resources required for the data source. By default, this will generate resources for SQL data source(s) in the
+   * context's `dataSourceStrategies` and `customSqlDataSourceStrategies`, but the generator can be invoked independently to support schemas
+   * with custom SQL directives but no models.
+   * @param context the TransformerContextProvider
+   * @param strategyOverride an optional override for the SQL database strategy to generate resources for.
+   */
+  generateResources(context: TransformerContextProvider, strategyOverride?: SQLLambdaModelDataSourceStrategy): void {
```

Comment:
Note: I moved the `isEnabled` check to an early exit condition and removed a layer of indentation from the bulk of the method. Viewing this diff with "hide whitespace" enabled shows the meaningful changes.